operator/apis/machinelearning.seldon.io/v1/seldondeployment_webhook.go

```diff
@@ -540,18 +547,6 @@ func (r *SeldonDeploymentSpec) ValidateSeldonDeployment() error {
 
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 
-// +kubebuilder:webhook:path=/mutate-machinelearning-seldon-io-v1-seldondeployment,mutating=true,failurePolicy=fail,sideEffects=None,groups=machinelearning.seldon.io,resources=seldondeployments,verbs=create;update,versions=v1,name=v1.mseldondeployment.kb.io
```

Comment:
All the code related to the webhook should be removed as well, as there are several functions in this file which would no longer be called

Reply:
We need to keep the mutating functionality which is called from Reconcile

Reply:
Ok makes sense, although it woudl still be worth moving it from the webhook file as it would no longer be used by the webhook itself

Reply:
Actually the code will be required in the webhooks -> validating webhooks were receiving defaulted resource before.
Now they need to apply defaulting on their own before validating -> I suspect that was  the reason why tests failed in previous run.